scripts/gen_mmu.py

```
@@ -0,0 +1,425 @@
+    # print(num_of_regions,hex(PD_start_addr))
+    for region in range(num_of_regions):
+        basic_mem_region_values = struct.unpack_from(struct_mmu_regions_format,
+                                               read_buff,
+                                               size_read_from_binary);
+        size_read_from_binary += struct.calcsize(struct_mmu_regions_format);
+
+        pde_index = get_pde_number(basic_mem_region_values[0])
+
+        # pte_valid_addr_start = (region_start_address >> 12) & 0x3FF;
+        pte_valid_addr_start = get_pte_number(basic_mem_region_values[0])
+
+        # make the end of the page table entry
+        pte_valid_addr_end = get_pte_number(basic_mem_region_values[0] +
+                                            basic_mem_region_values[1])
+
+        # if pte_valid_addr_end >= 1023:
+        #     pte_valid_addr_end =  1023
+
+        mem_size = basic_mem_region_values[1]
+        if basic_mem_region_values[1] > (1024*4096):
+            pte_valid_addr_end = 1024
+            # address can start from pte say 900 and then goto 1023 hence the size
+            mem_size = ( (pte_valid_addr_end - pte_valid_addr_start)*4096)
+
+        mem_region_values = mmu_region_details(start_addr = basic_mem_region_values[0],
+                                               size = mem_size,
+                                               pde_index = pde_index,
+                                               pte_valid_addr_start = pte_valid_addr_start,
+                                               pte_valid_addr_end = pte_valid_addr_end,
+                                               permissions = basic_mem_region_values[2])
+
+            # print(mem_region_values)
+        # print(hex(mem_region_values.start_addr),hex(mem_region_values.pde_index),
+        #       hex(mem_region_values.size))
+        if(mem_region_values.pde_index in list_of_pde.keys()):
+            validation_issue_memory_overlap = True
```

here you should throw an exception, with details on what regions are overlapping each other. i think this is better than raising an error in validate_pde_regions() because down there we won't know specifically which regions overlap each other.